src/cvar.ml

```diff
@@ -27,14 +27,15 @@ struct
   end
 
   let eval context t0 =
-    let res = ref Field.zero in
+    let open Field in
+    let res = of_int 0 in
     let rec go scale = function
-      | Constant c -> res := Field.add !res (Field.mul scale c)
-      | Var v -> res := Field.add !res (Field.mul scale (context v))
-      | Scale (s, t) -> go (Field.mul s scale) t
+      | Constant c -> res += mul scale c
+      | Var v -> res += mul scale (context v)
+      | Scale (s, t) -> go (mul s scale) t
```

Comment:
Can we use mutable multiplication for the scale too?

Reply:
This would mean we need an extra allocation for each add. I'll give it a try work a benchmark, but it seems like this might slow down additions like packing/unpacking